### DISCUSSION TITLE
Dwg Files in Assets - how to show them in Pimcore

### DISCUSSION THREAD

**Discussion by mircocervi** [pimcore/pimcore#8045] [General]
Hi all the group!
is there any plug to render dwg in Pimcore as an "image" assets?
The purpose would be to user "part" of the dwg draw and relate it to an object with "cut/focus" tools that are included in default DAM of Pimcore.
Thanks in adv
MC

**Comment by EliShklovsky** [comment]
Hi @mircocervi,
I am looking for such a solution too,
Did you managed to solve that?

**Comment by aarongerig** [comment]
@mircocervi @EliShklovsky AFAIK you cannot render a DWG in a browser by default. You'd have to convert the file format to something like `.glb` format and render it using something like [`<model-viewer>`](https://modelviewer.dev/). Should be doable, but doesn't exist yet.

**Comment by EliShklovsky** [comment]
Hi @aarongerig ,
Thanks for you answer.
I am looking for a preview to dwg cad file.
There is an open source CAD system: 
[QCAD](https://qcad.org/en/)
It has a command line tool like dwg2pdf, dwg2svg, dwg2bmp, which allow batch conversion of CAD drawing files.
I thought to hook one of the AssetEvent like pimcore.asset.preAdd and then create the preview file and load it to pimcore as a preview for the asset.
But I could not find any method to add a preview to an asset.

**Reply by brusch** [reply]
It seems Imagemagick is not supporting DWG, so currently it would be quite hard to integrate the support in Pimcore. But I agree it would be a nice addition,  PRs are welcome 😊

**Comment by EliShklovsky** [comment]
Hi @brusch ,
Is there a way in Pimcore to add a preview to an asset, like Pimcore do with imageimagick?
Because as wrote we can use the command line tool like QCAD to generate a preview for a dwg file.

**Reply by mircocervi** [reply]
I think we need a render for java/cad i am searching if I find something I will post here

**Reply by EliShklovsky** [reply]
Hi @mircocervi 
There is java/cad renderer:
https://qcad.org/
and it has a command line tool like:
./dwg2bmp [OPTIONS]... <drawing file>
Converts a drawing file (DWG, DXF, ...) to a bitmap file (BMP, PNG, ...).
I will appreciate if someone could lead me how to implement it to Pimcore
Thanks
Eli

**Comment by blankse** [comment]
We have also the problem with ".dwg" Files in the Assets. Pimcore declared it as Image because of the mime type "image/vnd.dwg".
There are many error log entries with it:
```
00:00:16 ERROR     [pimcore] Unable to load image: /var/www/html/public/var/assets/test.dwg  
00:00:16 ERROR     [pimcore] no decode delegate for this image format `DWG' @ error/constitute.c/ReadImage/564  
```
With the "Custom Asset Types" feature (https://github.com/pimcore/pimcore/pull/14310) we could create a new Asset Type for drawing files.

@mircocervi @EliShklovsky @aarongerig Does anyone of you already have a solution for this?

**Reply by aarongerig** [reply]
@blankse Unfortunately not. Would be cool though to have some sort of additional "asset type adapters", which handle previews of unsupported formats, e.g. DWG, INDD or the like.
